edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/Metric.cs

```diff
@@ -43,7 +44,7 @@ public bool Equals(Metric other)
             return this.TimeGeneratedUtc == other.TimeGeneratedUtc &&
                 this.Name == other.Name &&
                 this.Value == other.Value &&
-                this.Tags == other.Tags;
+                JsonConvert.SerializeObject(this.Tags) == JsonConvert.SerializeObject(other.Tags);
```

Comment:
don't use Json; just use dictionary count and except; or implement an extension method.

Reply:
I am using the following:
```
this.Tags.Count == other.Tags.Count && !this.Tags.Except(other.Tags).Any()
```